File: src/test/resources/dev/hilla/generator/basic/BasicEndpoint.ts
import client from '../../../../connect-client.default.js';
import type { EndpointRequestInit } from '@hilla/frontend';

async function fromArrayOfStringsToNothing(argument: string[], init?: EndpointRequestInit): Promise<void> {
    return client.call('dev.hilla.generator.basic.BasicEndpoint', 'fromArrayOfStringsToNothing', { argument }, init);
};

async function fromNothingToListOfStrings(init?: EndpointRequestInit): Promise<string[]> {
    return client.call('dev.hilla.generator.basic.BasicEndpoint', 'fromNothingToListOfStrings', {}, init);
};

async function fromPrimitiveToPrimitiveArray(argument: number, init?: EndpointRequestInit): Promise<number[]> {
    return client.call('dev.hilla.generator.basic.BasicEndpoint', 'fromPrimitiveToPrimitiveArray', { argument }, init);
};

async function fromStringToString(argument: string, init?: EndpointRequestInit): Promise<string> {
    return client.call('dev.hilla.generator.basic.BasicEndpoint', 'fromStringToString', { argument }, init);
};

async function fromTwoToMapOfStringsToList(init: string, end: string, _init?: EndpointRequestInit): Promise<Map<string, string[]>> {
    return client.call('dev.hilla.generator.basic.BasicEndpoint', 'fromTwoToMapOfStringsToList', { init, end }, _init);
};

const BasicEndpoint = {
    fromArrayOfStringsToNothing,
    fromNothingToListOfStrings,
    fromPrimitiveToPrimitiveArray,
    fromStringToString,
    fromTwoToMapOfStringsToList,
};

export default BasicEndpoint;

File: src/test/java/dev/hilla/generator/basic/BasicEndpoint.java
package dev.hilla.generator.basic;

import dev.hilla.parser.annotations.Endpoint;
import java.util.List;
import java.util.Map;

@Endpoint
public class BasicEndpoint {
  public void fromArrayOfStringsToNothing(String[] argument) {}

  public List<String> fromNothingToListOfStrings() {
    return List.of();
  }

  public int[] fromPrimitiveToPrimitiveArray(int argument) {
    return new int[] {argument};
  }

  public String fromStringToString(String argument) {
    return argument;
  }

  public Map<String, List<String>> fromTwoToMapOfStringsToList(String init, String end) {
    return Map.of();
  }
}
